lib/features/mailbox/data/network/mailbox_api.dart

```diff
@@ -401,6 +403,54 @@ class MailboxAPI with HandleSetErrorMixin {
     return listMailboxIdSubscribe ?? [];
   }
 
+  Map<String, List<String>?> _updateRightsForSubaddressing(MailboxSubaddressingAction action, Map<String, List<String>?>? currentRights) {
+    final updatedRights = Map<String, List<String>?>.from(currentRights ?? {});
+
+    updatedRights.update('anyone', (rights) {
+          if (action == MailboxSubaddressingAction.allow) {
+            rights?.add('r');
+          } else {
+            rights?.remove('r');
+          }
+          return rights;
+    },
+        ifAbsent: () => action == MailboxSubaddressingAction.allow ? ['r'] : null
+    );
+
+    return updatedRights;
+  }
+
+  Future<bool> subaddressingMailbox(Session session, AccountId accountId, SubaddressingMailboxRequest request) async {
+    final setMailboxMethod = SetMailboxMethod(accountId)
+      ..addUpdates({
+        request.mailboxId.id : PatchObject({
+          MailboxProperty.rights: _updateRightsForSubaddressing(request.subaddressingAction, request.currentRights)
+        })
+      });
+
+    final requestBuilder = JmapRequestBuilder(httpClient, ProcessingInvocation());
+
+    final setMailboxInvocation = requestBuilder.invocation(setMailboxMethod);
+
+    final capabilities = setMailboxMethod.requiredCapabilities
+        .toCapabilitiesSupportTeamMailboxes(session, accountId);
+
+    final response = await (requestBuilder
+      ..usings(capabilities))
+        .build()
+        .execute();
+
+    final setMailboxResponse = response.parse<SetMailboxResponse>(
+        setMailboxInvocation.methodCallId,
+        SetMailboxResponse.deserialize);
+
+    return Future.sync(() async {
+      return setMailboxResponse?.updated?.containsKey(request.mailboxId.id) ?? false;
+    }).catchError((error) {
+      throw error;
+    });
```

Comment:
we should not add try/catch at here. 
- should check condition and throw exception. 
- Should try/catch at DataSource

Reply:
you're right, there was already a try/catch block in MailboxDataSourceImpl